plugins/common/tls/config.go

```diff
@@ -49,7 +50,7 @@ func (c *ClientConfig) TLSConfig() (*tls.Config, error) {
 	// want TLS, this will require using another option to determine.  In the
 	// case of an HTTP plugin, you could use `https`.  Other plugins may need
 	// the dedicated option `TLSEnable`.
-	if c.TLSCA == "" && c.TLSKey == "" && c.TLSCert == "" && !c.InsecureSkipVerify {
+	if c.TLSCA == "" && c.TLSKey == "" && c.TLSCert == "" && !c.InsecureSkipVerify && c.ServerName == "" {
```

Comment:
I don't think this is correct. If you neither specify a CA nor a KEY or CERT but a `ServerName` then this check will not trigger but I doubt it makes sense to check the servername without validating the certificate!?

Reply:
The `ServerName` directive [does not affect only the host name in certificate checking](https://golang.org/src/crypto/tls/common.go#L588): It's also the hostname requested via SNI in the TLS [`ClientHello`](https://en.wikipedia.org/wiki/Server_Name_Indication#Technical_principles) message - that's the thing that I need to customize that prompted me to submit the PR in the first place (:

Even if that weren't the case, an empty TLS Config is [documented to use the host's root CA](https://golang.org/src/crypto/tls/common.go#L579) set if no root CAs are provided, so setting that value on an otherwise-empty TLS config would be meaningful.

Reply:
Could you please put a comment there linking to the docu you mentioned. Just for the next one asking himself why this should make sense!? Thanks for the explaination!

Reply:
Hm, so I re-read the TODO comment above & started auditing our code's usage of `TLSConfig()`, where I found out that the only case where a nil tls.Config matters (because the stdlib function tls.Client requires a non-nil value) is the x509_cert.go code... where we override the ServerName based on a configuration setting (incompatibly with the "common" ServerName setting now)! 

So I'll have to fix the ServerName usage in x509_cert.go, but more importantly: I think we should remove that TODO, as it could be a pretty meaningful distinction in terms of "were there any settings that the operator customized" (but ultimately, it looks like there's not much of a difference). What do you think?